sycl/source/detail/platform_impl.cpp

```diff
@@ -65,123 +74,18 @@ PlatformImplPtr platform_impl::getPlatformFromPiDevice(RT::PiDevice PiDevice,
   return getOrMakePlatformImpl(Plt, Plugin);
 }
 
-static bool IsBannedPlatform(platform Platform) {
-  // The NVIDIA OpenCL platform is currently not compatible with DPC++
-  // since it is only 1.2 but gets selected by default in many systems
-  // There is also no support on the PTX backend for OpenCL consumption,
-  // and there have been some internal reports.
-  // To avoid problems on default users and deployment of DPC++ on platforms
-  // where CUDA is available, the OpenCL support is disabled.
-  //
-  auto IsNVIDIAOpenCL = [](platform Platform) {
-    if (Platform.is_host())
-      return false;
-
-    const bool HasCUDA = Platform.get_info<info::platform::name>().find(
-                             "NVIDIA CUDA") != std::string::npos;
-    const auto Backend =
-        detail::getSyclObjImpl(Platform)->getPlugin().getBackend();
-    const bool IsCUDAOCL = (HasCUDA && Backend == backend::opencl);
-    if (detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_ALL) && IsCUDAOCL) {
-      std::cout << "SYCL_PI_TRACE[all]: "
-                << "NVIDIA CUDA OpenCL platform found but is not compatible."
-                << std::endl;
-    }
-    return IsCUDAOCL;
-  };
-  return IsNVIDIAOpenCL(Platform);
-}
-
 std::vector<platform> platform_impl::get_platforms() {
   std::vector<platform> Platforms;
-  const std::vector<plugin> &Plugins = RT::initialize();
-
-  info::device_type ForcedType = detail::get_forced_type();
-  for (unsigned int i = 0; i < Plugins.size(); i++) {
-
-    pi_uint32 NumPlatforms = 0;
-    // Move to the next plugin if the plugin fails to initialize.
-    // This way platforms from other plugins get a chance to be discovered.
-    if (Plugins[i].call_nocheck<PiApiKind::piPlatformsGet>(
-            0, nullptr, &NumPlatforms) != PI_SUCCESS)
-      continue;
-
-    if (NumPlatforms) {
-      std::vector<RT::PiPlatform> PiPlatforms(NumPlatforms);
-      if (Plugins[i].call_nocheck<PiApiKind::piPlatformsGet>(
-              NumPlatforms, PiPlatforms.data(), nullptr) != PI_SUCCESS)
-        return Platforms;
-
-      for (const auto &PiPlatform : PiPlatforms) {
-        platform Platform = detail::createSyclObjFromImpl<platform>(
-            getOrMakePlatformImpl(PiPlatform, Plugins[i]));
-        // Skip platforms which do not contain requested device types
-        if (!Platform.get_devices(ForcedType).empty() &&
-            !IsBannedPlatform(Platform))
-          Platforms.push_back(Platform);
-      }
-    }
+  RT::initialize();
+  std::vector<PlatformImplPtr> &PlatformCache =
```

Comment:
It seems platform mutex should be locked here.

Reply:
Thanks. Done.